src/platform/intel/cavs/memory.c

```diff
@@ -97,115 +99,111 @@ static struct block_map lp_buf_heap_map[] = {
-	.system[2] = {
-		.heap = HEAP_SYSTEM_2_BASE,
-		.size = HEAP_SYSTEM_2_SIZE,
-		.info = {.free = HEAP_SYSTEM_2_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-	.system[3] = {
-		.heap = HEAP_SYSTEM_3_BASE,
-		.size = HEAP_SYSTEM_3_SIZE,
-		.info = {.free = HEAP_SYSTEM_3_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-#endif
-	.system_runtime[0] = {
-		.blocks = ARRAY_SIZE(sys_rt_0_heap_map),
-		.map = sys_rt_0_heap_map,
-		.heap = HEAP_SYS_RUNTIME_0_BASE,
-		.size = HEAP_SYS_RUNTIME_0_SIZE,
-		.info = {.free = HEAP_SYS_RUNTIME_0_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE | SOF_MEM_CAPS_DMA,
-	},
-	.system_runtime[1] = {
-		.blocks = ARRAY_SIZE(sys_rt_x_heap_map),
-		.map = sys_rt_x_heap_map,
-		.heap = HEAP_SYS_RUNTIME_1_BASE,
-		.size = HEAP_SYS_RUNTIME_1_SIZE,
-		.info = {.free = HEAP_SYS_RUNTIME_1_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE | SOF_MEM_CAPS_DMA,
-	},
-#if defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE)
-	.system_runtime[2] = {
-		.blocks = ARRAY_SIZE(sys_rt_x_heap_map),
-		.map = sys_rt_x_heap_map,
-		.heap = HEAP_SYS_RUNTIME_2_BASE,
-		.size = HEAP_SYS_RUNTIME_2_SIZE,
-		.info = {.free = HEAP_SYS_RUNTIME_2_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-	.system_runtime[3] = {
-		.blocks = ARRAY_SIZE(sys_rt_x_heap_map),
-		.map = sys_rt_x_heap_map,
-		.heap = HEAP_SYS_RUNTIME_3_BASE,
-		.size = HEAP_SYS_RUNTIME_3_SIZE,
-		.info = {.free = HEAP_SYS_RUNTIME_3_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-#endif
-	.runtime[0] = {
-		.blocks = ARRAY_SIZE(rt_heap_map),
-		.map = rt_heap_map,
-		.heap = HEAP_RUNTIME_BASE,
-		.size = HEAP_RUNTIME_SIZE,
-		.info = {.free = HEAP_RUNTIME_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-	.buffer[0] = {
-		.blocks = ARRAY_SIZE(buf_heap_map),
-		.map = buf_heap_map,
-		.heap = HEAP_BUFFER_BASE,
-		.size = HEAP_BUFFER_SIZE,
-		.info = {.free = HEAP_BUFFER_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_EXT |
-			SOF_MEM_CAPS_CACHE,
-	},
-	.buffer[1] = {
-		.blocks = ARRAY_SIZE(hp_buf_heap_map),
-		.map = hp_buf_heap_map,
-		.heap = HEAP_HP_BUFFER_BASE,
-		.size = HEAP_HP_BUFFER_SIZE,
-		.info = {.free = HEAP_HP_BUFFER_SIZE,},
-		.caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_HP |
-			SOF_MEM_CAPS_CACHE | SOF_MEM_CAPS_DMA,
-	},
-/* To be removed if Apollolake gets LP memory*/
+struct mm memmap;
+
+void platform_init_memmap(void)
+{
+	int i;
+
+	/* .system vars */
+	uint32_t sys_base, sys_size, sys_size_t; // total sys size
+
+	/* .system_runtime vars*/
+	uint32_t sys_rt_base, sys_rt_size, sys_rt_size_t, sys_rt_blocks;
+	struct block_map *sys_rt_map;
+
+	/* .runtime vars */
+	uint32_t rt_base;
+
+	/* .system and .system_runtime initialization */
+	for (i = 0; i < PLATFORM_CORE_COUNT; i++) {
+		if (i == 0) { /* master core */
```

Comment:
How about moving core-0 related code before the loop, begin from 1 and remove if?

Reply:
done